src/psyclone/parse/algorithm.py

```diff
@@ -202,7 +203,7 @@ def parse(self, alg_filename):
                 "'{0}'".format(alg_filename))
 
         self._unique_invoke_labels = []
-        self._arg_name_to_module_name = {}
+        self._arg_name_to_module_name = OrderedDict()
```

Comment:
Because of the failure of the `test_invalid_builtin_kernel` test with Python 2. I updated the test example `15.12.1_invalid_builtin_kernel.f90` from
```
use inf,      only: field_type
```
to
```
use constants_mod, only: r_def
use field_mod,     only: field_type
```
and the error message `must either be named in a use statement (found ['constants_mod', 'field_mod'])` was not correct due to different order of `use` statements in Python 2 tests. I thought that `OrderedDict` would be good to use anyway as it preserves the order of `use` statements in the code.

Reply:
Great.